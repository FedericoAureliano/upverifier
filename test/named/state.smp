data state := state {round : Int}

function update(s : state) : state := 
  s.round := s.round + 1;
  (if s.round = 17 then
    s.round := s.round + 2;
    s.round := s.round + 3
  else
    s.round := s.round + 4;
    (if s.round >= 30 then
      s.round := s.round - 30
    else 
      s.round := s.round);
    s.round := s.round + 5);
  s.round := s.round + 6

function start : state := state(0)

print(start.round)
print(update(start).round)
print(update(update(start)).round)
print(update(update(update(start))).round)