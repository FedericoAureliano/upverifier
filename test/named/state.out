0
16
28
14
